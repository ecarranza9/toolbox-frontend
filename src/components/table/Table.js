import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useSelector } from 'react-redux';
import { Container, Form, Spinner, Table } from 'react-bootstrap';
import { getFilesData, getFilesList, setFileName } from '../../redux/slices/files';
import { formattedData } from '../../helpers';

const CustomTable = () => {

  const dispatch = useDispatch();
  const { files, isFetching, list, fileName } = useSelector(state => state.files);

  const dataToTable = formattedData(files);

  const handleSelect = (e) => {
    dispatch(setFileName(e.target.value));
  };

  useEffect(
    () => {
      dispatch(getFilesData());
      dispatch(getFilesList());
    }, [dispatch]);

  useEffect(() => {
    if (fileName && fileName.length) {
      dispatch(getFilesData(fileName));
    }
  }, [dispatch, fileName]);


  return (
    <Container style={{ marginTop: 20 }}>
      <Form.Select
        aria-label="Select File"
        data-testid="select-files"
        style={{ marginBottom: 30 }} onChange={handleSelect}
      >
        <option value='all'>All files data</option>
        {list && list.length > 0 && list.map(file => (
          <option value={file}>{file}</option>
        ))}
      </Form.Select>
      {
        isFetching ? (
          <Spinner animation="border" role="status" style={
            { display: 'flex', marginLeft: '50%', marginTop: 60 }
          } >
            <span className="visually-hidden">Loading...</span>
          </Spinner>
        )
          : dataToTable.length > 0 ? (
            <div data-testid="table">
              <Table striped bordered hover>
                <thead>
                  <tr>
                    <th>FileName</th>
                    <th>Text</th>
                    <th>Number</th>
                    <th>Hex</th>
                  </tr>
                </thead>
                <tbody>
                  {dataToTable && dataToTable.map((item, index) => (
                    <tr key={index}>
                      <td>{item.file}</td>
                      <td>{item.text}</td>
                      <td>{item.number}</td>
                      <td>{item.hex}</td>
                    </tr>
                  ))}
                </tbody>
              </Table>
            </div>
          ) : <span> Este archivo no tiene datos validos </span>}
    </Container >
  );
};

export default CustomTable;
