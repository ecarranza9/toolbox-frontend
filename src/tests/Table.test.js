import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Table from '../components/table/Table';
import { Provider } from 'react-redux';
import filesReducer from '../redux/slices/files';
import { configureStore } from '@reduxjs/toolkit';

const renderWithRedux = (
  component,
  { initialState, store = configureStore({ reducer: filesReducer }) } = {
    list: [],
    files: [],
    isFetching: false
  }
) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  }
}

afterEach(cleanup);

describe('Table Component', () => {
  const { asFragment } = renderWithRedux(<Table />)
  test('should match snapshot and styles', () => {
    expect(asFragment(<Table />)).toMatchSnapshot();
  });
});