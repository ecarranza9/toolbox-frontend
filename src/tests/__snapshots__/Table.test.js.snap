// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should match snapshot and styles 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="margin-top: 20px;"
  >
    <select
      aria-label="Select File"
      class="form-select"
      style="margin-bottom: 30px;"
    >
      <option
        value="all"
      >
        All files data
      </option>
    </select>
    <table
      class="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            FileName
          </th>
          <th>
            Text
          </th>
          <th>
            Number
          </th>
          <th>
            Hex
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
